src/activities/Elsa.Activities.Http/Formatters/XMLContentFormatter.cs

```diff
@@ -0,0 +1,123 @@
+    {
+        public static List<Type> TypeCache { get; set; }
+    }
+
+
+    public class XMLContentFormatter : IContentFormatter
+    {
+        public int Priority => 1;
+        public IEnumerable<string> SupportedContentTypes => new[] { "application/xml", "text/xml" };
+
+        public Task<object> ParseAsync(byte[] content, string contentType)
+        {
+
+            GetTypeCache();
+
+            string rootNodeTypeName = GetRootTypeName(ref content);
+
+            if (rootNodeTypeName != "")
+            {
+                Type rootType = GetRootType(rootNodeTypeName);
+                XmlSerializer xmlSerializer = new XmlSerializer(rootType);
+                var model = xmlSerializer.Deserialize(new MemoryStream(content));
+                return Task.FromResult<object>(model);
+            }
+
+            // this is not clear - what will be returned if theres no deserialization?
+            return null;
+        }
+
+
+        public Task<byte[]> ParseAsync(object content, string contentType)
+        {
+            Type rootType = content.GetType();
+            XmlSerializer xmlSerializer = new XmlSerializer(rootType);
+            using (MemoryStream stream = new MemoryStream())
+            {
+                xmlSerializer.Serialize(stream, content);
+                return Task.FromResult<byte[]>(stream.ToArray());
+            }
+        }
+
+        private static Type GetRootType(string rootNodeName)
+        {
+            return XMLContentFormatterCache.TypeCache.Where(t => t.Name == rootNodeName).FirstOrDefault();
+        }
+
+        private static void GetTypeCache()
+        {
+            if (XMLContentFormatterCache.TypeCache == null)
+            {
+                string baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+                foreach (string fileName in Directory.EnumerateFiles(baseDirectory, "*.dll", SearchOption.AllDirectories))
+                {
+                    AssemblyLoadContext.Default.LoadFromAssemblyPath(Path.Combine(baseDirectory, fileName));
+                }
+
+                List<string> elsaLibs =
+                    AppDomain.CurrentDomain.GetAssemblies()
+                    .Where(a => a.GetName().Name.StartsWith("Elsa"))
+                    .Select(a => a.FullName).ToList();
+
+                elsaLibs.AddRange(
+                    AppDomain.CurrentDomain.GetAssemblies()
+                    .Where(a => a.GetName().Name.StartsWith("Elsa"))
+                    .SelectMany(a => a.GetReferencedAssemblies())
+                    .Select(a => a.FullName).ToList());
+
+
+
+                XMLContentFormatterCache.TypeCache =
+                    AppDomain.CurrentDomain.GetAssemblies().
+                    Where(a =>
+                        {
+                            string name = a.GetName().Name;
+                            return !name.StartsWith("System") &&
+                                    !name.StartsWith("Microsoft") &&
```

Comment:
This logic seems way too specific and not open for extensibility.

Reply:
Can be removed if we supply the wanted binary and the ModelType